### PR TITLE
Add `baseURL` parameter to `GACAppCheckDebugProvider` constructor

### PR DESCRIPTION
**Pull request by andrewheard**
Added a `baseURL` parameter to `GACAppCheckDebugProvider` that offers the same functionality as the one in [`GACAppAttestProvider`](https://github.com/google/app-check/blob/f77879ca7581666d9fe0da7b5b27c34196acd902/AppCheckCore/Sources/Public/AppCheckCore/GACAppAttestProvider.h#L40-L48). This optionally allows the `https://firebaseappcheck.googleapis.com/v1beta` endpoint to be used instead of the default (`https://firebaseappcheck.googleapis.com/v1`).

### CODE REVIEW

**Review by ncooke3 — APPROVED** [review]
If it wasn't considered:

I see that the two URLs are the same except for the last component–
```
https://firebaseappcheck.googleapis.com/v1beta
https://firebaseappcheck.googleapis.com/v1
```
Is it preferable for GACAppcheck to save `https://firebaseappcheck.googleapis.com` and clients to only configure the "BackendVersionURLComponent" (rather than whole URL)? No need to bother with such a refactor unless you hadn't considered it and think it's in some way better than current approach.

**Comment by andrewheard** [comment]
> If it wasn't considered:
> 
> I see that the two URLs are the same except for the last component–
> 
> ```
> https://firebaseappcheck.googleapis.com/v1beta
> https://firebaseappcheck.googleapis.com/v1
> ```
> 
> Is it preferable for GACAppcheck to save `https://firebaseappcheck.googleapis.com` and clients to only configure the "BackendVersionURLComponent" (rather than whole URL)? No need to bother with such a refactor unless you hadn't considered it and think it's in some way better than current approach.

@ncooke3 At the moment the approach you mentioned would be feasible (we are currently only using `v1` or `v1beta` on the prod endpoints) but I have a slight preference towards keeping the full URL. If we want to quickly test a fix or feature before it reaches prod, we could easily specify alternate endpoints like `staging-foo.sandbox.googleapis.com` or `autopush-foo.sandbox.googleapis.com` (I don't remember the exact endpoints for App Check but they differ in the subdomain part of the URL).

tldr; the full URL is more flexible though it is a lot more verbose.

**Comment by ncooke3** [comment]
Makes sense to me. LGTM!
